lib/ansible/modules/cloud/vmware/vmware_guest_video.py

```diff
@@ -0,0 +1,325 @@
+            vm_obj: Managed object of virtual machine
+        Returns: Video Card device and a list of dict video card configuration
+        """
+        video_facts = dict()
+        video_card = None
+        for device in vm_obj.config.hardware.device:
+            if isinstance(device, vim.vm.device.VirtualVideoCard):
+                video_card = device
+                video_facts = dict(
+                    auto_detect=device.useAutoDetect,
+                    display_number=device.numDisplays,
+                    video_memory=device.videoRamSizeInKB,
+                    enable_3D_support=device.enable3DSupport,
+                    renderer_3D=device.use3dRenderer,
+                    memory_3D=device.graphicsMemorySizeInKB,
+                )
+                break
+        return video_card, video_facts
+
+    def get_video_card_spec(self, vm_obj):
+        """
+        Get device changes of virtual machine
+        Args:
+            vm_obj: Managed object of virtual machine
+        Returns: virtual device spec
+        """
+        video_card, video_card_facts = self.gather_video_card_facts(vm_obj)
+        self.video_card_facts = video_card_facts
+        if video_card is None:
+            self.module.fail_json(msg='Not get video card device of specified virtual machine.')
+        video_spec = vim.vm.device.VirtualDeviceSpec()
+        video_spec.operation = vim.vm.device.VirtualDeviceSpec.Operation.edit
+        video_spec.device = video_card
+        auto_detect = False
+        enabled_3d = False
+
+        if self.params['gather_video_facts']:
+            return None
+        if self.params['use_auto_detect'] is not None:
+            if video_card_facts['auto_detect'] and self.params['use_auto_detect']:
+                auto_detect = True
+            elif not video_card_facts['auto_detect'] and self.params['use_auto_detect']:
+                video_spec.device.useAutoDetect = True
+                self.change_detected = True
+                auto_detect = True
+            elif video_card_facts['auto_detect'] and not self.params['use_auto_detect']:
+                video_spec.device.useAutoDetect = False
+                self.change_detected = True
+        else:
+            if video_card_facts['auto_detect']:
+                auto_detect = True
+        # useAutoDetect set to False then display number and video memory config can be changed
+        if not auto_detect:
+            if self.params['display_number'] is not None:
+                if self.params['display_number'] < 1:
+                    self.module.fail_json(msg="display_number attribute valid value: 1-10.")
+                if self.params['display_number'] != video_card_facts['display_number']:
+                    video_spec.device.numDisplays = self.params['display_number']
+                    self.change_detected = True
+
+            if self.params['video_memory_mb'] is not None:
+                if self.params['video_memory_mb'] < 2:
```

Comment:
If the check is actually for 2Mb, then the message should state that, i feel (including the module documentation). Otherwise, the check should be 
```suggestion
                if float(self.params['video_memory_mb']) < 1.17:
```

Reply:
The warning message from vCenter UI is "1.172 to 128 or 256", and it accepts float input on the UI, but in the background I see that is transformed to int. So here I want to accept int. I'll modify this. Thanks.